nixos/modules/config/update-users-groups.pl

```diff
@@ -235,7 +235,11 @@ sub parseUser {
     # Ensure home directory incl. ownership and permissions.
     if ($u->{createHome} and !$is_dry) {
         make_path($u->{home}, { mode => oct($u->{homeMode}) }) if ! -e $u->{home};
-        chown $u->{uid}, $u->{gid}, $u->{home};
+        if ($u->{recursiveCreateHome}) {
+            system("chown", "-R", "$u->{uid}:$u->{gid}", $u->{home});
+        } else {
+            chown $u->{uid}, $u->{gid}, $u->{home};
+        }
```

Comment:
Although unlikely to fail, it is missing some checking of exit code, and having two ways of doing chown seems unnecessary imho.




Suggestion:

```suggestion
        system( "chown", 
                ($u->{recursiveCreateHome} ? "-R" : ()),
                "$u->{uid}:$u->{gid}", 
                $u->{home} ) == 0 || die "Unable to chown $u->{home}: $!";
```

(Replace `die` with `warn` if this error doesnt need to be fatal)

Reply:
Thank you for your review :heart: 
> and having two ways of doing chown seems unnecessary imho.

yeah that's fair enough.
> (Replace die with warn if this error doesnt need to be fatal)

Even if this was a fatal error I would still set this to warn because it happens in stage one of boot and other steps of the script include stuff like setting the user password, and I would want the user to still be able to login and read the error without having to boot into a rescue image.